test/extensions/filters/http/jwt_authn/authenticator_test.cc

```diff
@@ -136,6 +136,43 @@ TEST_F(AuthenticatorTest, TestCompletePaddingInJwtPayload) {
   EXPECT_EQ(headers.get_("sec-istio-auth-userinfo"), ExpectedPayloadValueWithPadding);
 }
 
+// This test verifies whether the claim is successfully added to header or not
+TEST_F(AuthenticatorTest, TestClaimToHeader) {
+  createAuthenticator();
+  EXPECT_CALL(*raw_fetcher_, fetch(_, _))
+      .WillOnce(Invoke([this](Tracing::Span&, JwksFetcher::JwksReceiver& receiver) {
+        receiver.onJwksSuccess(std::move(jwks_));
+      }));
+
+  Http::TestRequestHeaderMapImpl headers{
+      {"Authorization", "Bearer " + std::string(NestedGoodToken)}};
+
+  expectVerifyStatus(Status::Ok, headers);
+
+  EXPECT_EQ(headers.get_("x-jwt-claim-sub"), "test@example.com");
+  EXPECT_EQ(headers.get_("x-jwt-claim-nested"), "value1");
+  EXPECT_EQ(headers.get_("x-jwt-bool-claim"), "true");
+  EXPECT_EQ(headers.get_("x-jwt-int-claim"), "9999");
+}
+
+// This test verifies when wrong claim is passed in claim_to_header
+TEST_F(AuthenticatorTest, TestClaimToHeaderWithHeaderReplace) {
+  createAuthenticator();
+  EXPECT_CALL(*raw_fetcher_, fetch(_, _))
+      .WillOnce(Invoke([this](Tracing::Span&, JwksFetcher::JwksReceiver& receiver) {
+        receiver.onJwksSuccess(std::move(jwks_));
+      }));
+
+  Http::TestRequestHeaderMapImpl headers{
```

Comment:
I did not see a claim header passed-in here.

We need a test to verify a pass-in claim header is replaced with a good token,  and is removed with a bad token.

Reply:
That part is in the verification, this test assumes that jwt is already verified and header sanitization wouldn't be affected. I am testing that logic here - https://github.com/envoyproxy/envoy/blob/96addf2784fe8da718945325688db77711512701/test/extensions/filters/http/jwt_authn/provider_verifier_test.cc#L203

Reply:
I see. thanks